File: Convert_Types/Octal_Decimal.java
import java.util.Scanner;

public class Octal_Decimal {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter an octal number: ");
        int octal = sc.nextInt();

        int decimal = 0;
        int power = 1;
        int rem;

        while (octal > 0) {
            rem = octal % 10;
            decimal += rem * power;
            power *= 8;
            octal /= 10;
        }

        System.out.println("Decimal Equivalent = " + decimal);
    }
}

File: Convert_Types/Octal_Decimal.c
#include <stdio.h>
void mian()
{
    int octal;

    printf("Enter an octal number: ");
    scanf("%d", &octal);

    int decimal = 0, power = 1, rem = 0;

    while (octal > 0)
    {
        rem = octal % 10;
        decimal += rem * power;
        power *= 8;
        octal /= 10;
    }

    printf("Decimal Equivalent = %d",decimal);
}